src/category.rs

```diff
@@ -1,57 +1,242 @@
 use toxml::ToXml;
-use error::Error;
 
 /// A representation of the `<category>` element.
 #[derive(Debug, Default, Clone, PartialEq)]
-pub struct Category {
+pub struct Category
+{
     /// The name of the category.
-    pub name: String,
+    name: String,
     /// The domain for the category.
-    pub domain: Option<String>,
+    domain: Option<String>,
 }
 
-impl FromXml for Category {
+impl Category
+{
+    /// Get the category that exists under `Category`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rss::{CategoryBuilder, Category};
+    ///
+    /// let category = "podcast";
+    ///
+    /// let category_obj = CategoryBuilder::new()
+    ///     .name(category)
+    ///     .finalize()
+    ///     .unwrap();
+    ///
+    /// assert_eq!(category, category_obj.name().as_str());
+    /// ```
+    pub fn name(&self) -> String
+    {
+        self.name
+            .clone()
+    }
+
+    /// Get the optional domain that exists under `Category`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rss::{CategoryBuilder, Category};
+    ///
+    /// let domain_string = "http://jupiterbroadcasting.com/".to_owned();
+    ///
+    /// let category = CategoryBuilder::new()
+    ///     .domain(Some(domain_string.clone()))
+    ///     .finalize()
+    ///     .unwrap();
+    ///
+    /// let domain_option = category.domain();
+    /// assert_eq!(Some(domain_string.clone()), domain_option);
+    /// ```
+    ///
+    /// ```
+    /// use rss::{CategoryBuilder, Category};
+    ///
+    /// let category = CategoryBuilder::new()
+    ///     .domain(None)
+    ///     .finalize()
+    ///     .unwrap();
+    ///
+    /// let domain_option = category.domain();
+    /// assert!(domain_option.is_none());
+    /// ```
+    pub fn domain(&self) -> Option<String>
+    {
+        self.domain
+            .clone()
+    }
+}
+
+impl FromXml for Category
+{
     fn from_xml<R: ::std::io::BufRead>(mut reader: XmlReader<R>,
                                        element: Element)
-                                       -> Result<(Self, XmlReader<R>), Error> {
+        -> Result<(Self, XmlReader<R>), Error>
+    {
         let mut domain = None;
 
-        for attr in element.attributes().with_checks(false).unescaped() {
+        for attr in element.attributes()
+                           .with_checks(false)
+                           .unescaped() {
             if let Ok(attr) = attr {
                 if attr.0 == b"domain" {
-                    domain = Some(try!(String::from_utf8(attr.1.into_owned())));
+                    domain = Some(String::from_utf8(attr.1
+                                                        .into_owned())?);
                     break;
                 }
             }
         }
 
         let content = element_text!(reader).unwrap_or_default();
 
-        Ok((Category {
-            name: content,
-            domain: domain,
-        }, reader))
+        Ok((Category { name: content,
```

Comment:
By default rustmft puts the first value on a new line which I think looks better. My preference would be to respect that style.